sycl/test/esimd/on-device/PrefixSum.cpp

```diff
@@ -0,0 +1,470 @@
+//==----------- PrefixSum.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/esimd.hpp>
+#include <iostream>
+
+#define MAX_TS_WIDTH 1024
+#define TUPLE_SZ 1 // kernel can only handle TUPLE_SZ can be 1, 2, 4
+
+#if TUPLE_SZ == 1
+#define GATHER_SCATTER_MASK ESIMD_R_ENABLE
+#elif TUPLE_SZ == 2
+#define GATHER_SCATTER_MASK ESIMD_GR_ENABLE
+#elif TUPLE_SZ == 4
+#define GATHER_SCATTER_MASK ESIMD_ABGR_ENABLE
+#endif
+
+#define LOG_ENTRIES 8
+#define PREFIX_ENTRIES (1 << LOG_ENTRIES)
+#define PREFIX_ENTRIES_LOW 32
+#define ENTRIES_THRESHOLD 2048
+#define MIN_NUM_THREADS                                                        \
+  1 // minimum number of threads to launch a kernel (power of 2)
```

Comment:
This is no good. The macro will expand into `1 // and the comment` which may lead to unexpected and unwanted consequences.

Reply:
Fixed.